### PR TITLE
REF: de-nest Series.__setitem__

### PR DESCRIPTION
**Pull request by jbrockmendel**
Splitting this into a couple of steps for readability.  The two main things I have in mind here are a) de-nesting, b) try to parallel the structure of `__getitem__`

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -1045,13 +1047,13 @@ def _set_with(self, key, value):
             else:
                 key_type = lib.infer_dtype(key, skipna=False)
 
+            # Note: key_type == "boolean" should not occur because that
```
Comment (jreback) [inline pandas/core/series.py]: can you add an assert for this

**Review by jreback — CHANGES_REQUESTED** [review]
comment, rebase and ping on green.

**Comment by jbrockmendel** [comment]
ping

File: pandas/core/series.py (hunk 1 above)
Comment (jreback) [reply pandas/core/series.py]: did you want to add this here?

Comment (jbrockmendel) [reply pandas/core/series.py]: hah, looks like i put the assertion with the corresponding comment in _get_with, not here in _set_with.  removed the assertion for symmetry.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. can you rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**